src/components/App/UrlForm.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { ChangeEvent, FormEvent } from 'react';
+import { FaLink } from 'react-icons/fa6';
+
+import Button from '@/components/Button';
+
+interface ErrorMessageProps {
+    message: string;
+}
+
+const ErrorMessage: React.FC<ErrorMessageProps> = ({ message }) => {
+    return (
+        <p className="text-black font-semibold text-base mt-1 w-full flex transition-opacity duration-500 ease-in-out">
+            {message}
+        </p>
+    );
+};
+
+interface UrlFormProps {
+    url: string;
+    error: string | null;
+    clearError: () => void;
+    setUrl: (url: string) => void;
+    onSubmit: (url: string) => void;
+    loading: boolean;
+}
+
+const UrlForm: React.FC<UrlFormProps> = ({ url, setUrl, onSubmit, error, clearError, loading }) => {
+    const handleSubmit = (e: FormEvent) => {
+        e.preventDefault();
+        const formData = new FormData(e.target as HTMLFormElement);
+        const url = formData.get('URL') as string | null;
+
+        if (!url) return;
+
+        onSubmit(url);
+    };
+
+    const handleUrlChange = (e: ChangeEvent<HTMLInputElement>) => {
+        setUrl(e.target.value);
+        clearError();
+    };
+    const inputBorderClass = error ? 'border-2 border-red-500' : '';
+    const buttonIcon = loading ? (
+        <div
+            className="inline-block h-4 w-4 animate-spin rounded-full border-2 border-solid border-current border-e-transparent align-[-0.125em] text-surface motion-reduce:animate-[spin_1.5s_linear_infinite] dark:text-gray-600"
+            role="status"
+        >
+            <span className="!absolute !-m-px !h-px !w-px !overflow-hidden !whitespace-nowrap !border-0 !p-0 ![clip:rect(0,0,0,0)]">
+                Loading...
+            </span>
+        </div>
+    ) : (
+        <FaLink />
+    );
```

Comment:
- should we add a loading state to the button instead as it will allow us to easily toggle the state on and off wherever the button component is used?

Reply:
fixed